hicexplorer/hicExport.py

```diff
@@ -197,7 +197,7 @@ def main(args=None):
             hic_ma.matrix = (triu(hic_ma.matrix, k=-limit) - triu(hic_ma.matrix, k=limit)).tocsr()
             hic_ma.matrix.eliminate_zeros()
 
-    if not args.inputFormat == 'cool' and args.chromosomeOrder is not None and len(args.chromosomeOrder) == 1:
+    if args.inputFormat != 'cool' and args.chromosomeOrder is not None and len(args.chromosomeOrder) > 0:
```

Comment:
You probably wish to remove this line (and unindent the block that follows). This condition is problematic, and not necessary. E.g. if the format is "h5" and `len(args.chromosomeOrder) == 1`, you still want to execute the block. Now it is not the case (even after this fix).

Reply:
Here the data is already loaded, and we don't need to bother about the features of cooler format anymore. Just need to apply `args.chromosomeOrder` and `args.clearMaskedBins` accordingly, no matter what the format etc. was. I propose to remove this line.

Reply:
We need to run this line but different than it is right now: 
If we can partially load data with cooler we only have one region or chromosome, there is nothing to select or sort. In the case there are two or more chromosomes / regions we need it, independent if it was cooler or something else. 

Therefor I propose:
```
if args.chromosomeOrder is not None and len(args.chromosomeOrder) > 1:
    hic_ma.keepOnlyTheseChr(args.chromosomeOrder)
if args.clearMaskedBins:
    hic_ma.maskBins(hic_ma.nan_bins)
```

Reply:
What about the following case: format "h5", `len(args.chromosomeOrder) == 1`? You then wish to call `keepOnlyTheseChr`, isn't it?

Reply:
True. 
```
if (args.chromosomeOrder is not None and len(args.chromosomeOrder) > 1) or (args.inputFormat is not 'cool' and args.chromosomeOrder is not None):
```

`len(args.chromosomeOrder)  > 0` is implicitly given if `args.chromosomeOrder is not None` says the documentation: https://docs.python.org/2/library/argparse.html#nargs

Reply:
The condition now makes sense. :) Actually it can be rewritten a bit shorter:
```
if args.chromosomeOrder is not None and (args.inputFormat is not 'cool' or len(args.chromosomeOrder) > 1):
```

Reply:
Even shorter:

```
if args.chromosomeOrder  and (args.inputFormat is not 'cool' or len(args.chromosomeOrder) > 1):
```
after this, line 201 is not needed.